pygmt/src/filter1d.py

```diff
@@ -0,0 +1,144 @@
+        the initial sign test; in that case the filter will return 0.0.
+        Apart from custom coefficients (**f**), the other filters may accept
+        variable filter widths by passing *width* as a two-column time-series
+        file with filter widths in the second column.  The filter-width file
+        does not need to be co-registered with the data as we obtain the
+        required filter width at each output location via interpolation.  For
+        multi-segment data files the filter file must either have the same
+        number of segments or just a single segment to be used for all data
+        segments.
+
+    end : bool
+        Include Ends of time series in output. The default [False] loses
+        half the filter-width of data at each end.
+
+    time_col : int or str
+        Indicates which column contains the independent variable (time). The
+        left-most column is 0, while the right-most is (*n_cols* - 1)
+        [Default is 0].
+
+    output_type : str
+        Determine the format the xyz data will be returned in [Default is
+        ``pandas``]:
+
+            - ``numpy`` - :class:`numpy.ndarray`
+            - ``pandas``- :class:`pandas.DataFrame`
+            - ``file`` - ASCII file (requires ``outfile``)
+    outfile : str
+        The file name for the output ASCII file.
+
+    Returns
+    -------
+    ret : pandas.DataFrame or numpy.ndarray or None
+        Return type depends on ``outfile`` and ``output_type``:
+
+        - None if ``outfile`` is set (output will be stored in file set by
+          ``outfile``)
+        - :class:`pandas.DataFrame` or :class:`numpy.ndarray` if ``outfile`` is
+          not set (depends on ``output_type`` [Default is
+          :class:`pandas.DataFrame`])
+
+    """
+    if kwargs.get("F") is None:
+        raise GMTInvalidInput("Pass a required argument to 'filter'.")
+    if output_type not in ["numpy", "pandas", "file"]:
+        raise GMTInvalidInput("Must specify format as either numpy, pandas, or file.")
+    if outfile is not None and output_type != "file":
+        msg = (
+            f"Changing `output_type` of filter1d from '{output_type}' to 'file' "
+            "since `outfile` parameter is set. Please use `output_type='file'` "
+            "to silence this warning."
+        )
+        warnings.warn(msg, category=RuntimeWarning, stacklevel=2)
+        output_type = "file"
+    elif output_type == "file" and outfile is None:
+        raise GMTInvalidInput("Must specify outfile for ASCII output.")
+
+    with GMTTempFile() as tmpfile:
+        with Session() as lib:
+            file_context = lib.virtualfile_from_data(check_kind="vector", data=data)
+            with file_context as infile:
+                if outfile is None:
+                    outfile = tmpfile.name
+                lib.call_module(
+                    "filter1d", build_arg_string(kwargs, infile=infile, outfile=outfile)
```

Comment:
Following the style in #1910.
```suggestion
                    module="filter1d", 
                    args=build_arg_string(kwargs, infile=infile, outfile=outfile),
```